python/src/scipp/plotting/figure.py

```diff
@@ -145,19 +146,27 @@ def initialize(self, axformatters):
                 ticker.LogLocator()
             }
 
-    def connect(self, controller, event_handler):
+    def connect(self, controller):
         """
         Connect the toolbar to callback from the controller. This includes
         rescaling the data norm, and change the scale (log or linear) on the
         axes.
         """
         if self.toolbar is not None:
             self.toolbar.connect(controller=controller)
-        self.fig.canvas.mpl_connect('button_press_event',
-                                    event_handler.handle_button_press)
-        self.fig.canvas.mpl_connect('pick_event', event_handler.handle_pick)
-        self.fig.canvas.mpl_connect('motion_notify_event',
-                                    event_handler.handle_motion_notify)
+
+    def toggle_profile_connection(self, visible, event_handler):
```

Comment:
The figure (and view) don't need to know anything about profiles, this should be called `toggle_mouse_events` or something like that, I think.

Also the name of the parameter (`visible`) should be changed in that case.